### PR TITLE
On Windows, always use reg.exe from System32 directory

### PR DESCRIPTION
**Pull request by teobugslayer**
- [x] @fresc81
 - [x] @justinmchase
 - [x] @mikeyoon

See https://github.com/fresc81/node-winreg/issues/17

### CODE REVIEW

File: lib/registry.js [hunk 1]
```diff
@@ -118,7 +119,17 @@ util.inherits(RegistryItem, Object);
...
 Object.freeze(RegistryItem.prototype);
 
-
+/**
+ * Get the path to system's reg.exe. Useful when another reg.exe is added to the PATH
+ * Implemented only for Windows
+ */
+function getRegExePath() {
+    if (process.platform === 'win32') {
+        return path.join(process.env.windir, 'system32', 'reg.exe');
```
Comment (justinmchase) [inline lib/registry.js]: Do you think its worth checking for the files existence?

Comment (fresc81) [inline lib/registry.js]: not really. child_process.spawn() should handle cases where the executable is missing and reflect the error using #25 .

Comment (fresc81) [inline lib/registry.js]: I am fine with this solution. I still only want to check if the arch option works properly. I guess I have some time this weekend.

Comment (mikeyoon) [inline lib/registry.js]: In my really brief testing it looks like /reg:32/64 is backwards compatible in Win7 32-bit (just always returns the 32-bit hive).

**Comment by teobugslayer** [comment]
@justinmchase,

reg.exe is subject to [Windows File Protection](https://support.microsoft.com/en-us/kb/222193) since Windows 2000, so it would be extremely hard for the file to disappear.

**Comment by RX14** [comment]
@teobugslayer Regarding your comment that file exists checks are pointless, this is certainly untrue. The time between the file exists check and it being used is so short that the chance of it being deleted in that small time is incredibly tiny. Node 5 deprecates that API in favour of better ones, not because it is useless.

**Comment by justinmchase** [comment]
Ok let's not debate the merits of checking files before attempting to read them here :)

I was merely thinking, if `reg.exe` wasn't at the default place then changing the behavior to use standard path rules might make sense. But based on @teobugslayer's comments it should be assumable that the file is always there and if its not then you probably have a big problem anyway. So in that case, I think its fine to assume the file location and not have backup behavior.

**Comment by fresc81** [comment]
Works fine, thank you for your contribution.

**Comment by teobugslayer** [comment]
When can we expect to release this in the npm?

**Comment by fresc81** [comment]
I just published it.

**Comment by teobugslayer** [comment]
Great news, thank you! I'll update to it as soon as possible.

**Pull request merged**